### PR TITLE
bazel: add forbidigo as a linter and make it runnable by nogo

### PR DESCRIPTION
**Pull request by burmudar**
Adds forbidigo as a linter than can be executed by nogo.

### What are these other BUILD files for?
The build didn't work without them so had to add them. They were auto generated by `bazel run //:gazelle`

### Example of errors reported by this linter
```
external/com_github_prometheus_common/model/time.go:164:10: use of `fmt.Errorf` forbidden by pattern `^fmt\.Errorf$` (forbidigo)
external/com_github_prometheus_common/model/time.go:196:13: use of `fmt.Errorf` forbidden by pattern `^fmt\.Errorf$` (forbidigo)
external/com_github_prometheus_common/model/time.go:200:13: use of `fmt.Errorf` forbidden by pattern `^fmt\.Errorf$` (forbidigo)
external/com_github_prometheus_common/model/value.go:53:10: use of `fmt.Errorf` forbidden by pattern `^fmt\.Errorf$` (forbidigo)
external/com_github_prometheus_common/model/value.go:279:10: use of `fmt.Errorf` forbidden by pattern `^fmt\.Errorf$` (forbidigo)
external/com_github_prometheus_common/model/value.go:327:10: use of `fmt.Errorf` forbidden by pattern `^fmt\.Errorf$` (forbidigo)
[609 / 1,838] 10 actions running
    GoCompilePkg external/com_github_yuin_goldmark_emoji/definition/definition.a; 3s darwin-sandbox
    GoCompilePkg external/com_github_go_enry_go_enry_v2/data/data.a; 3s darwin-sandbox
ERROR: build interrupted
INFO: Elapsed time: 10.957s, Critical Path: 4.42s
INFO: 403 processes: 101 internal, 302 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

### Why do we have our own runAnalysis
Annoyingly, the `analyzer.NewAnalyzer` from forbidigo does not allow you to configure it in anyway, and we want to add our patterns into it. So `runAnalysis` was copied from `forbidigo` and slightly modified so that we can specify patterns

## Test plan
`bazel build //cmd/frontend/...`

### CODE REVIEW

**Review by jhchabran — APPROVED** [review]

File: dev/linters/forbidigo/forbidigo.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}{
+	IgnorePermitDirective: true,
+	ExcludeGodocExamples:  true,
+	AnalyzeTypes:          true,
+}
+
+func init() {
+	// We replace run here with our own runAnalysis since the one from NewAnalyzer
+	// doesn't allow us to specify patterns ...
+	Analyzer.Run = runAnalysis
```
Comment (burmudar) [inline dev/linters/forbidigo/forbidigo.go:33]: I guess we could just create the analyzer on line 13 ourselves and set run there ...

**Comment by jhchabran** [comment]
@burmudar you have a linter failure within the linter wrapping code itself!

**Comment by burmudar** [comment]
> @burmudar you have a linter failure within the linter wrapping code itself!

Thanks I'll take a look!

**Comment by burmudar** [comment]
> > @burmudar you have a linter failure within the linter wrapping code itself!
> 
> Thanks I'll take a look!

Fixed:
- run `bazel run //:gazelle -- fix`
- add config to ignore `rules_go` ... for some reason it was linting this dir
- build:
- - `//cmd/...`
- - `//internal/...`
- - `//enterprise/...`

**Pull request merged**